jtsync/jtsync.rb

```diff
@@ -0,0 +1,135 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ENV["CURRENT_GEMFILE"] ||= File.expand_path("Gemfile", __FILE__)
+
+require "bundler"
+Bundler.setup(:default)
+require "trello"
+
+
+class Failed < StandardError
+end
+
+
+def board
+  @trello_board ||= if ENV.has_key? "TRELLO_BOARD"
+                      Trello::Board.find(ENV["TRELLO_BOARD"])
+                    else
+                      Trello::Board.find("ywSwlQpZ")
+                    end
+end
+
+
+def state_from_env
+  case ENV["JOB_STATUS"]
+  when "0" then 'successful'
+  else 'failed'
+  end
+end
+
+
+# Handling different naming conventions between opensuse/suse
+# and matrix/normal job
+#
+# returns tuple: (list_name, job_name)
+def job_from_env
+  # is this a ci.opensuse.org matrix job? (Juno)
+  if ENV.has_key? "openstack_project"
+    return ["OpenStack", "#{ENV["JOB_NAME"].gsub("openstack-", "")}: #{ENV["openstack_project"]}"]
+  end
+
+  # is this a ci.suse.de matrix job? (Devel:Cloud:7:Staging)
+  if ENV.has_key? "project"
+    version = ENV["project"].split(":")[2]
+    raise Failed.new("Can not determine ci.suse.de matrix job name") if version.nil?
+
+    return ["Cloud #{version}", "C#{version} #{ENV["JOB_NAME"]}"]
+  end
+
+  #normal job from ci.suse.de (cloud-mkcloud7-job-4nodes-linuxbridge-x86-64)
+  version = ENV["JOB_NAME"].split("-")[1]
+
+  raise Failed.new("Cannot determin ci.suse.de job name") if version.nil?
+
+  ["Cloud #{version[-1,1]}", ENV["JOB_NAME"]]
+end
+
+
+def notify_card_members(card)
+  members = card.members.map do |m|
+    "@" + Trello::Member.find(m.id).username
+  end
+
+  comment = card.add_comment("#{members.join(" ")}: card status changed")
+  comment_id = JSON.parse(comment)["id"]
+
+  card.comments.select do |c|
+    c.action_id == comment_id
+  end.map(&:delete)
+end
+
+
+def update_card_label(card, state)
+  label = board.labels.select {|l| l.name == state }.first
+  raise Failed.new("Could not find label '#{state}'") if label.nil?
+
+  return if card.labels.include? label
+
+  card.labels.each{|l| card.remove_label(l) }
+  card.add_label(label)
+end
+
+
+def find_card_for(job)
+  list_name, job_name = job
+
+  list = board.lists.select {|l| l.name == list_name }.first
+  raise Failed.new("Could not find list #{list_name}") if list.nil?
+
+  card = list.cards.select {|c| c.name == job_name }.first
+  raise Failed.new("Could not find card matching #{job_name} in #{list_name}") if card.nil?
+
+  card
+end
+
+
+# run the script --------------------------------------------------------------
+begin
+  ["TRELLO_PUBLIC_KEY", "TRELLO_MEMBER_TOKEN", "JOB_NAME", "JOB_STATUS"].each do |required|
+    unless ENV.has_key? required
+      puts("Environment variable is missing: #{required}")
+      exit 0
+    end
+  end
+
+  Trello.configure do |config|
+    config.developer_public_key = ENV["TRELLO_PUBLIC_KEY"]
+    config.member_token = ENV["TRELLO_MEMBER_TOKEN"]
```

Comment:
@Itxaka 
If done right this could work of course. Its more a way to prevent leaks that way by removing the possibility.
Furthermore we already deploy a `.netrc` on the worker hosts other services, so its easy to also add the credentials for trello there. So they will not be read by the script, they don't need to handled explicitly and errors are less likely.